--- a/marshmallow_dataclass/__init__.py
+++ b/marshmallow_dataclass/__init__.py
@@ -396,6 +396,8 @@
         if hasattr(v, "__marshmallow_hook__") or k in MEMBERS_WHITELIST
     }
 
+    include_non_init = getattr(getattr(clazz, "Meta", None), "include_non_init", False)
+
     # Update the schema members to contain marshmallow fields instead of dataclass fields
     type_hints = get_type_hints(
         clazz, localns=clazz_frame.f_locals if clazz_frame else None
@@ -412,7 +414,7 @@
             ),
         )
         for field in fields
-        if field.init
+        if field.init or include_non_init
     )
 
     schema_class = type(clazz.__name__, (_base_schema(clazz, base_schema),), attributes)
